### ISSUE TITLE
x265 encoding settings lost when muxing raw HEVC stream to MKV

### ISSUE THREAD

**Issue by a-roussos** [mbunkus/mkvtoolnix#1984]
I'm using the MKVToolNix GUI v11.0.0 64bit on Windows 7 64bit. When muxing a raw HEVC stream (created with x265 v2.4) to MKV, information such as the x265 version and the encoding settings used is not kept (I'm using MediaInfo GUI v0.7.95 to view such details).

However, when using the previous MKVToolNix GUI version (v10.0.0), all works well.

FWIW, a similar issue was reported a while back (#1076).

**Comment by mbunkus** [comment]
I'm pretty sure this has already been fixed. Please try the latest [pre-builds](https://mkvtoolnix.download/windows/pre/).

**Comment by a-roussos** [comment]
Thanks for the quick reply @mbunkus. I just tried muxing with the latest pre-build and it works fine. Cheers!

**Comment by mbunkus** [comment]
For the record: this was fixed by the same commit that #1958 was fixed by.

**Comment by vahiddddd** [comment]
Thank you.

On May 13, 2017 1:12 PM, "Andreas Roussos" <notifications@github.com> wrote:

I'm using the MKVToolNix GUI v11.0.0 64bit on Windows 7 64bit. When muxing
a raw HEVC stream (created with x265 v2.4) to MKV, information such as the
x265 version and the encoding settings used is not kept (I'm using
MediaInfo GUI v0.7.95 to view such details).

However, when using the previous MKVToolNix GUI version (v10.0.0), all
works well.

FWIW, a similar issue was reported a while back (#1076
<https://github.com/mbunkus/mkvtoolnix/issues/1076>).

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/mbunkus/mkvtoolnix/issues/1984>, or mute the thread

**Comment by rarbg** [comment]
@mbunkus 
and what do we do if we have h265 raw muxed with 10.0 that we want to fix ? with hex editor data is present , but if an already muxed file with 10.0 is re-muxed with 11.0 it doesnt fix it

**Comment by mbunkus** [comment]
There's not really anything you can do to fix a broken file. You'll have to re-mux from a good one.

**Comment by rarbg** [comment]
@mbunkus 
It actually plays fine both audio video and subtitles. Also md5 frame verification passes for each frame. Can you define whats broken ?

**Comment by mbunkus** [comment]
The SPS contains superfluous bits somewhere near the end (but not right at the end). These bits are present due to some unfortunate code duplication in the SPS handling routines which emit a certain section twice, but only under certain circumstances. Depending on the actual SPS settings, this may or may not cause parsers to abort due to invalid data. It's also nothing that can be detected unambiguously: the only thing that can be determined is that something is wrong, but not exactly where in the SPS. Not even that fact can be detected in some circumstances as SPS are padded with filler bits to the next byte boundary, and if the superfluous bits all fit within those filler bits, then a parser might not even see something wrong.

So basically all I can say is: if playback works, great, just keep hoping other players will keep playing them. If it doesn't work, then using the broken copy any further is not a good idea, and remuxing from a known-good source is the solution I advice you to use.

**Comment by rarbg** [comment]
Thanks for the detailed explanation

**Comment by sneaker2** [comment]
@rarbg 
> and what do we do if we have h265 raw muxed with 10.0 that we want to fix ? with hex editor data is present , but if an already muxed file with 10.0 is re-muxed with 11.0 it doesnt fix it

This regression affects 11.0.0, not 10.0.0.

>Not even that fact can be detected in some circumstances as SPS are padded with filler bits to the next byte boundary, and if the superfluous bits all fit within those filler bits, then a parser might not even see something wrong.

I think x264 even uses such filler bits to insert a pattern to make it possible to detect x264 has been used.

**Comment by rarbg** [comment]
@sneaker2 
oops that was a typo. anyways i already made a simple script that fixed those
